File: 프로그래머스/연습문제/Lv1/명예의 전당 (1)/명예의 전당 (1).py
def solution(k, score):
    answer=[]; scores=[]
    for sc in score:
        if len(scores) < k: scores.append(sc)
        elif sc > min(scores):
            scores.pop(scores.index(min(scores)))
            scores.append(sc)

        answer.append(min(scores))

    return answer

File: 프로그래머스/연습문제/Lv1/명예의 전당 (1)/명예의 전당 (1).cpp
#include <string>
#include <vector>
#include <queue>
using namespace std;

vector<int> solution(int k, vector<int> score) {
    priority_queue<int, vector<int>, greater<int>> pq;
    vector<int> answer;
    
    for(int sc : score){
        if(pq.size() < k) pq.push(sc);
        else if(sc > pq.top()){
            pq.pop();
            pq.push(sc);
        }
        
        answer.push_back(pq.top());
    }
    
    return answer;
}
